include/circt/Dialect/Moore/MooreOps.td

```diff
@@ -146,7 +146,40 @@ def VariableOp : MooreOp<"variable", [
   let arguments = (ins StrAttr:$name, Optional<UnpackedType>:$initial);
   let results = (outs Res<UnpackedType, "", [MemAlloc]>:$result);
   let assemblyFormat = [{
-    `` custom<ImplicitSSAName>($name) ($initial^)? attr-dict
+    custom<ImplicitSSAName>($name) ($initial^)? attr-dict
+    `:` type($result)
+  }];
+}
+
+def NetOp : MooreOp<"net", [
+  DeclareOpInterfaceMethods<OpAsmOpInterface, ["getAsmResultNames"]>,
+  OptionalTypesMatchWith<"assigned value and variable types match",
+    "result", "assignment", "$_self">,
+]> {
+  let summary = "A net declaration";
+  let description = [{
+    The `moore.net' operation is a net declaration. Net types defines different types
+    of net connection in SV. There are twelve built-in net types defined in the official
+    standard construct of the operation:
+    `supply0`, `supply1`, `tri`, `triand`, `trior`, `trireg`, `tri0`, `tri1`, `uwire`,
+    `wire`, `wand`, `wor`.
```

Comment:
Since these twelve net types are part of the SV spec, could we represent them as an enum attribute instead of a string, for a less stringly-typed IR. Something like `I32EnumAttr` or, if it exists, even `I8EnumAttr` could do the trick. I think we had something like that floating around for port directions in the original PR.